src/librbd/librbd.cc

```diff
@@ -6352,6 +6352,47 @@ extern "C" ssize_t rbd_aio_compare_and_write(rbd_image_t image, uint64_t off,
   return 0;
 }
 
+extern "C" ssize_t rbd_aio_comparev_and_writev(rbd_image_t image,
+                                               uint64_t off, size_t len,
+                                               const struct iovec *cmp_iov,
+                                               int cmp_iovcnt,
+                                               const struct iovec *iov,
+                                               int iovcnt,
+                                               rbd_completion_t c,
+                                               uint64_t *mismatch_off,
+                                               int op_flags)
+{
+  librbd::ImageCtx *ictx = (librbd::ImageCtx *)image;
+  librbd::RBD::AioCompletion *comp = (librbd::RBD::AioCompletion *)c;
+  tracepoint(librbd, compare_and_write_enter, ictx, ictx->name.c_str(),
+             ictx->snap_name.c_str(),
+             ictx->read_only, off, len, NULL, NULL, op_flags);
+
+  size_t cmp_len;
+  int r = get_iovec_length(cmp_iov, cmp_iovcnt, cmp_len);
+  if (r != 0 || cmp_len < len) {
+    tracepoint(librbd, compare_and_write_exit, -EINVAL);
+    return -EINVAL;
+  }
+
+  size_t write_len;
+  r = get_iovec_length(iov, iovcnt, write_len);
+  if (r != 0 || write_len < len) {
+    tracepoint(librbd, compare_and_write_exit, -EINVAL);
+    return -EINVAL;
+  }
+
+  auto aio_completion = get_aio_completion(comp);
+  auto bl = iovec_to_bufferlist(ictx, iov, iovcnt, aio_completion);
+  auto cmp_bl = iovec_to_bufferlist(ictx, cmp_iov, cmp_iovcnt, aio_completion);
+  librbd::api::Io<>::aio_compare_and_write(
+    *ictx, aio_completion, off, len, std::move(cmp_bl), std::move(bl),
+    mismatch_off, op_flags, false);
```

Comment:
formatting (I realize that you just cut-and-pasted it)
```suggestion
  librbd::api::Io<>::aio_compare_and_write(*ictx, aio_completion, off, len,
                                           std::move(cmp_bl), std::move(bl),
                                           mismatch_off, op_flags, false);
```